corehq/apps/userreports/sql/adapter.py

```diff
@@ -180,14 +180,14 @@ def save_rows(self, rows):
             {i.column.database_column_name.decode('utf-8'): i.value for i in row}
             for row in rows
         ]
-        if self.session_helper.is_citus_db:
+        if self.session_helper.is_citus_db and not reassigning_cases:
```

Comment:
One alternative here, is to have this use `last_supervisor_id` rather than `supervisor_id` (or both to be extra safe in case you are concerned a case was already updated by the pillow). That would maintain the performance improvements this provides while ensuring the delete was safe for citus.